src/components/core/PivotTable.tsx

```diff
@@ -493,9 +547,63 @@ export class PivotTableInner extends BaseVisualization<IPivotTableInnerProps, IP
             (!alreadyResized() || (alreadyResized() && force))
         ) {
             this.resizing = true;
-            setTimeout(() => {
-                this.autoresizeVisibleColumns(event.columnApi, previouslyResizedColumnIds);
-            }, AGGRID_BEFORE_RESIZE_TIMEOUT);
+            await this.autoresizeVisibleColumns(event.columnApi, previouslyResizedColumnIds);
+            await this.growToFit(event.columnApi);
+            this.resizing = false;
+            await this.sleep(AGGRID_ON_RESIZE_TIMEOUT);
+            this.setState({
+                resized: true,
+            });
+        }
+    };
+
+    private resetColumnsWidthToDefault(
+        columnApi: ColumnApi,
+        columns: Column[],
+        previouslyResizedColumnIds: string[] = [],
+    ) {
+        columns.forEach(col => {
+            const id = this.getColumnIdentifier(col.getDefinition() as IGridHeader);
+            const resizedColumnId = previouslyResizedColumnIds.find(i => i === id);
+
+            if (resizedColumnId) {
+                if (
```

Comment:
Please, create for this condition variable or function. It should be more clear then.

Reply:
Done